OpenAdaptAI/OpenAdapt
# Fix tray signals

**What kind of change does this PR introduce?**
This PR fixes the issues with signals not being passed around properly on windows



**Summary**
This PR replaces the QSocketNotifier with a QThread and a worker that uses a signal in addition to a parent/child pipe because in windows to pass data around. This additional maneouver was needed because windows wasn't triggered the activated signal of the qsocketnotifier object. 

The current data flow is: `child_conn` -> `parent_conn` -> `signal emit` -> `signal emit listener`.

The signal emit listener is needed, because processing things in the Qthread object is not possible because of how Qt restricts updates from amywhere else other than the main thread. The signal emit listener is executed on the main thread, so the signal is used as a connection to send the data back to the main thread.



**Checklist**

**How can your code be run and tested?**
Run the code (in development mode) in windows, and ensure that you see the notifications and the tray text changes.




**Other information**
